### ISSUE TITLE
 Error uploading file: conanmanifest.txt, '404 Client Error: Not Found for url: https://artifactory.com/artifactory/api/conan/conan-test/v2/conans/xxx/revisions/b9c4826ad9445ea5257f6432b5a019dd/files/conanmanifest.txt[bug]

### ISSUE THREAD

**Issue by rajesh-kareti-infovision** [conan-io/conan#16537]
### Describe the bug

Asp part of automation uploading multiple packages. seeing this error randomly
 Error uploading file: conanmanifest.txt, '404 Client Error: Not Found for url: https://artifactory.com/artifactory/api/conan/conan-test/v2/conans/<pkg_name>/revisions/b9c4826ad9445ea5257f6432b5a019dd/files/conanmanifest.txt

As other packages uploading successfully there should not be any access issue. This run builds and uploads the package for multiple build_types

**Comment by czoido** [comment]
Hi @rajesh-kareti-infovision,

Thanks for the question, could you please provide more details on what you are doing so we can try to reproduce? Like Conan version, OS, Artifactory version, what is you upload flow, do you upload in paralell? etc.

**Comment by rajesh-kareti-infovision** [comment]
I am building and uploading  my package for different build_types Release , Debug , RelWithDebInfo. I am running these builds in parallel through github actions
conan version : 2.0.16
artifactory version : 7.55.17 rev 75517900
os : ubuntu 20.04

**Comment by czoido** [comment]
Hi @rajesh-kareti-infovision,

I think this could be an Artifactory issue with parallel uploads, probably a race condition, I think this could be a duplicate of https://github.com/conan-io/conan/issues/14581 and https://github.com/conan-io/conan/issues/14175 Please, have a look at them, you may find some valuable information there.

One thing you can try is avoiding the parallel upload of the recipe, limiting it to one of the CI jobs, uploading only the recipe, then the rest of binaries can be uploaded in parallel, as they will not compete for writing in the same recipe location. That may work.

I would also recommend to update your Conan version if that's possible to the latest one.

Hope this helps.

**Comment by memsharded** [comment]
There is some further discussion in https://github.com/conan-io/conan/issues/17211 ticket, with some recommendations how to do parallel uploads at the moment. 

Then, the ticket to track the possible future implementation of atomic uploads is this one: https://github.com/conan-io/conan/issues/16703, I am closing this ticket, please follow that other one. 

Thanks for your feedback!
